code/game/objects/items/RCD.dm

```diff
@@ -554,28 +573,63 @@ GLOBAL_VAR_INIT(icon_holographic_window, init_holographic_window())
-		//if we are trying to build a window on top of a grill [only if there isnt already a window there] then we skip this check because thats normal behaviour
-		if(rcd_results["mode"] == RCD_WINDOWGRILLE && locate(/obj/structure/grille, target_turf) && (!locate(/obj/structure/window, target_turf) && !locate(/obj/structure/window/reinforced, target_turf)))
-			//no checks proceed
+		//if we are trying to build a window on top of a grill then we skip this check because thats normal behaviour
+		if(rcd_results["mode"] == RCD_WINDOWGRILLE && istype(A, /obj/structure/grille))
+			//if a player builds a grill on top of himself manually with rods he can't finish the window if he is still in the grill.
+			var/list/structures_to_ignore
+			if(window_type == /obj/structure/window/fulltile || window_type == /obj/structure/window/reinforced/fulltile) //if we are trying to build an fultile window we only ignore the grill but other directional windows on the grill can block its construction
+				structures_to_ignore = list(/obj/structure/grille)
+			else //for normal directional windows we ignore other directional windows as they can be in diffrent directions on the grill. There is a later check during construction to deal with those
+				structures_to_ignore = list(/obj/structure/grille, /obj/structure/window, /obj/structure/window/reinforced)
+			if(is_turf_blocked(the_turf = target_turf, ignore_mobs = FALSE, ignored_atoms = structures_to_ignore))
+				playsound(loc, 'sound/machines/click.ogg', 50, TRUE)
+				balloon_alert(user, "something is on the grille!")
+				qdel(rcd_effect)
+				return FALSE
+
+		/**
+		 * if we are trying to create plating on turf which is not a proper floor then dont check for objects on top of the turf just allow that turf to be converted into plating. e.g. create plating beneath a player or underneath a machine frame/any dense object
+		 * if we are trying to finish a wall girder then let it finish
+		 */
+		else if(rcd_results["mode"] == RCD_FLOORWALL && (!istype(target_turf, /turf/open/floor) || istype(A, /obj/structure/girder)))
+			//if a player builds a wallgirder on top of himself manually with iron sheets he can't finish the wall if he is still on the girder. Exclude the girder itself when checking for other dense objects on the turf
+			if(istype(A, /obj/structure/girder) && is_turf_blocked(the_turf = target_turf, ignore_mobs = FALSE, ignored_atoms = list(/obj/structure/girder)))
+				playsound(loc, 'sound/machines/click.ogg', 50, TRUE)
+				balloon_alert(user, "something is on the girder!")
+				qdel(rcd_effect)
+				return FALSE
+
+		//turf density check
 		else
 			//structures which are small enough to fit on turfs containing directional windows.
 			var/static/list/small_structures = list(
+				RCD_WINDOWGRILLE,
 				RCD_MACHINE,
 				RCD_COMPUTER,
 				RCD_REFLECTOR,
 				RCD_FLOODLIGHT,
-				RCD_FURNISHING
+				RCD_FURNISHING,
+				RCD_AIRLOCK
```

Comment:
```suggestion
				RCD_AIRLOCK,
```

DM Supports Trailing Commas. Also, please alphabetize this list while you're doing this (there's a few extensions for VSC that will easily do this for you).